GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/IncludedFolderTests.cs

```diff
@@ -0,0 +1,262 @@
+
+            this.gvfsProcess.RemoveIncludedFolders("Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(folderPath);
+            fileSystemEntries.Length.ShouldEqual(6);
+            fileToCreate.ShouldBeAFile(this.fileSystem);
+
+            this.fileSystem.DeleteFile(fileToCreate);
+        }
+
+        [TestCase, Order(8)]
+        public void ModifiedFileShouldNotAllowIncludedFolderChange()
+        {
+            string modifiedPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.WriteAllText(modifiedPath, "New Contents");
+
+            string output = this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            output.ShouldContain("Include was aborted");
+            this.ValidateIncludedFolders(new string[0]);
+        }
+
+        [TestCase, Order(9)]
+        public void ModifiedFileAndCommitThenChangingIncludeFoldersShouldKeepFileAndFolder()
+        {
+            string modifiedPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.WriteAllText(modifiedPath, "New Contents");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "add .");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -m Test");
+
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldBeADirectory(this.fileSystem);
+            modifiedPath.ShouldBeAFile(this.fileSystem);
+        }
+
+        [TestCase, Order(10)]
+        public void CreateNewFileAndCommitThenRemoveIncludedFolderShouldKeepFileAndFolder()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder, "Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string fileToCreate = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "newfile.txt");
+            this.fileSystem.WriteAllText(fileToCreate, "New Contents");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "add .");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -m Test");
+
+            this.gvfsProcess.RemoveIncludedFolders("Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(folderPath);
+            fileSystemEntries.Length.ShouldEqual(2);

```

Comment:
Is this 2 because "RunFunctionalTests.bat" from the previous test is still there? (If so could you add a comment?)

Also, would the behavior be the same if the new file were created in a folder that did not previously have a modified file in it?  (should that be its own test?)

Reply:
Yeah I updated this to use a different path to remove that confusion.